watcher/watcher.go

```diff
@@ -118,8 +116,8 @@ func (wh *WatchHandler) GetWlidsToContainerToImageIDMap() WlidsToContainerToImag
 	return wh.wlidsToContainerToImageIDMap
 }
 
-func labelsToInstanceID(labels map[string]string) (string, error) {
-	instanceID, ok := labels[instanceIDAnnotationKey]
+func annotationsToInstanceID(annotations map[string]string) (string, error) {
```

Comment:
can't we make this function more generic (for every key we expect to be on annotations)? Or this is just temporary until we have an implementation in another pkg?

Reply:
It’s a combination of both. Ideally, we want to get instance IDs by calling a method on the object we receive, like ` SBOMSPDXv2p3.InstanceID()`. But this might take more effort than we are prepare to spend, like re-generating the APIServer code, updating all the dependents etc. Instead, we are adding this small function to the Operator for now.

The next thing is that the Operator needs to get only the instance ID. It doesn’t care for other keys, so why would we spend extra time to write tests and implementation for other keys? It’s not like we are exporting this method and someone else is going to use it. So I think this is enough for us.